kratos/python_scripts/process_factory.py

```diff
@@ -28,16 +30,16 @@ def ConstructListOfProcesses( self, process_list ):
 
                 try:
                     full_module_name = kratos_module_name + "." + python_module_name
-                    python_module = __import__(full_module_name,fromlist=[python_module_name])
+                    python_module = importlib.import_module(full_module_name)
                 except ImportError: # old import mechanism for backwards-compatibility
-                    __import__(kratos_module_name)
-                    python_module = __import__(python_module_name)
+                    importlib.import_module(kratos_module_name)
+                    python_module = importlib.import_module(python_module_name)
```

Comment:
why do you change this? 
It took me some time to figure this out. Also I think the behavior of importlib is depending on the version of python (at least some of the functionalities for sure)
I think this is a very sensitive change that should be discussed separately

Reply:
Didn't we agree on changing this last thursday?
I was under the impresion that we were using __import__ because of importlib being incomatible iwt <=2.6 version of python but concluding that it was fine to change as everybody was using it.

Reply:
ah ok that was that
Why is this better again than `__import__`?

I suggest we make a separate PR then we can update the apps also in the same PR, what do you think?

Reply:
Ok to me.

Basically importlib allows cx_freeze, nuitca, etc, detect dynamic imports, while calling __importlib__ does not. I think they hook the call and push the name of the lib in the module path so they know what to pack (or something like that, havn't studied the mechanism in detail)

Reply:
I'm not sure about the mechanism but as @roigcarlo said this change is needed for making dynamic imports work with cx_freeze and other packagers. As we have discussed it (and agreed informally) in the committee I asked  @roigcarlo to add it to this PR to have it a complete pack.

Reply:
I did not remember 
Either way, see #5198

Reply:
No problem,

As long as there are no conflict between this one and #5198 we are ok with separating it!

Reply:
No conflict, in fact I also ported this to most apps

Reply:
I see now. Many thanks!! :+1: